### ISSUE TITLE
Installer - Windows - UAC elevation

### ISSUE THREAD

**Issue by PRabahy** [bitcoin/bitcoin#5798]
For me the 0.10.0 installer didn't request UAC elevation leading to an unexpected behavior. First, I double clicked the installer from a standard user account. It begins the install process, but failed with "Error opening file for writing: C:\Program Files\Bitcoin\bitcoin-qt.exe[...]". I aborted that install attempt.

Next, I right clicked on the installer, selected Run as Administrator. That install worked fine, but on the last page, it automatically launches bitcoin-qt for you. I was surprised (I guess I should have expected this) because the instance of bitcoin-qt created its profile under the administrator account (including a new wallet.dat).

tl;dr - The windows Bitcoin installer should be UAC aware.

**Comment by theuni** [comment]
I don't see these issues with the 0.10 binaries on my win7 VM. The install process works as expected, and auto-launched bitcoin-qt runs as the user who ran the installer.

What win version are you running? Does 0.9.x have the same issues?

**Comment by PRabahy** [comment]
When you ran it in your win7 VM, did you have UAC enabled and were you using a non-admin account to launch the installer?

I am running Windows 7 64bit. Yes, 0.9.x has the same issue, but I just started running as a standard user recently.

**Comment by theuni** [comment]
Ok, this turns out to be a pain in the ass.

I created an installer that requests 'admin' elevation: https://bitcoincore.org/cfields/bitcoin-0.10.0/bitcoin-0.10.0-win64-uactest-setup.exe

It appears to work as intended, except that it actually installs for the elevated account rather than the current one. The results are not at all obvious.. "run bitcoin-qt now" from the installer runs as the admin user. After that, running the .exe from program files (no shortcuts are installed for the current user) prompts for a data path. Probably exactly what you saw when running the official setup.exe as Administrator.

It appears as though non-trivial changes to the installer would be needed.

It's been quite a while since I've used a Windows environment, do most applications fare well in this scenario?

**Comment by PRabahy** [comment]
A lot of other installers fall into the same UAC trap(s) as bitcoin.

I'm not sure if having the installer always request elevation is the right thing for the reasons you mentioned. The easiest "fix" IMO would be to always request elevation and remove the "run bitcoin-qt now screen".

Ideally the installer should only request elevation if it actually needs it. For example, if I install to "C:\Program Files\Bitcoin" it will need to elevate, but if I install to "C:\Users\PRabahy\Bitcoin", it doesn't. If the installer has admin privileges, it can put shortcuts in the "All Users" folder, but if it doesn't then just put them in the current user folder. It should also "run bitcoin-qt now" as the user that did the install (not the user that did the elevation).

The ideal installer would do the following steps:
1. User runs the installer
2. User picks the install location
3. Installer check if it needs to elevate
3a. If so, the installer requests elevation
4. The install is preformed
5. Install shortcuts to All Users if possible, if not install shortcuts to Current User.
6. If elevated, de-elevate.
7. Show the "run bitcoin-qt now" screen.

**Comment by PRabahy** [comment]
I just ran the Armory installer and noticed that it follows a different option for elevated vs non-elevated. If I run it as a standard user, it defaults the install directory to C:\Users\PRabahy\AppData\Local\Armory, but if I run it as an elevated user, it default to C:\Program Files (x86)\Armory. At least this doesn't give an error when I double click the installer.

**Comment by laanwj** [comment]
Right: It should be possible to install Bitcoin Core locally for a user, so without admin elevation. There have been attempts to "fix" this before which made that impossible.

**Comment by sipsorcery** [comment]
@MarcoFalke maybe another one for closing.

I just tested the v0.20.0 install on Windows 10 and there are no UAC prompts (and thus no requirement for admin permissions) either during the install or when subsequently executing bitcoin-qt.

I doubt this issue will warrant looking at the installers for previous versions.

**Comment by MarcoFalke** [comment]
Is this still an issue with a recent version of Bitcoin Core? If yes, what are the steps to reproduce?

**Comment by trianglesis** [comment]
Win 10
bitcoin-0.21.0-win64-setup.exe
Still having this issue.
I do not want to crack the security of my PC to only run this. There're a lot of workarounds playing with policies and accounts. which I don't want to corrupt.

**Comment by achow101** [comment]
@trianglesis You are likely running into a different issue, see https://github.com/bitcoin-core/gui/issues/252

**Comment by trianglesis** [comment]
Looks like the latest win package from the current repo is working fine with a single warning before run.
